src/driver.rs

```diff
@@ -9,18 +9,37 @@ use std::{
     io::{self, Read, Write},
 };
 
-/// process first filepath in the input Vec, or stdin if empty
-pub fn process_files(filepaths: Vec<String>) -> Result<i32> {
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// A lexer, parser, interpreter, JIT and AOT compiler and
+/// runtime written in Rust for a new (simple) language
+#[derive(StructOpt, Debug)]
+#[structopt(name = "mylang")]
+pub struct Opt {
+    /// Files to process
+    #[structopt(name = "FILE", parse(from_os_str))]
+    files: Vec<PathBuf>,
+}
+
+/// process first filepath in Files, or stdin if empty
+///
+/// # Errors
+///
+/// Will return Err if files couldn't be opened, or found
+/// or if they could not be lexed, parsed or evaluated
+pub fn go(opt: &Opt) -> Result<i32> {
```

Comment:
I wonder if there's a more informative name for this function?

Reply:
`drive`? it's the real `main` -- the bit that lives in the library instead of the binary